File: previousJava/src/model/StadePhysio.java
package model;

public enum StadePhysio {

		
		ADULTE("Adulte", 0),
		CROISSANCE("Croissance", 1),
		LACTATION("Lactation", 2),
		GESTATION("Gestation", 3),
		HOSPIT("Hospit", 4)
		
		
		;


	private String name = "";


	private int categorie=0;

	 
	//Constructeur
	 StadePhysio(String name, int categrorieo){
	  this.name = name;

	  this.categorie=categrorieo;

	}
	 
	 public String nameToString() {
		return name;
	}
	 public int getCategorie() {
		return categorie;
	}
	 public static String getStringFromInt(int id){
		 String str=StadePhysio.ADULTE.nameToString();
		 for (StadePhysio espe:StadePhysio.values()){
			 if (id==espe.getCategorie()){
				 str=espe.nameToString();
			 }
		 }
		 return str;
	 }
	 public static StadePhysio getEnumFromInt(int id){
		 StadePhysio esp=StadePhysio.ADULTE;
		 for (StadePhysio espe:StadePhysio.values()){
			 if (id==espe.getCategorie()){
				 esp=espe;
			 }
		 }
		 return esp;
	 }
	 public static StadePhysio getEnumFromString(String id){
		 StadePhysio esp=StadePhysio.ADULTE;
		 for (StadePhysio espe:StadePhysio.values()){
			 if (id.equals(espe.nameToString())){
				 esp=espe;
			 }
		 }
		 return esp;
	 }

	 




}

File: composeApp/src/commonMain/kotlin/fr/vetbrain/vetnutri_mp/Enumer/StadePhysio.kt
package fr.vetbrain.vetnutri_mp.Enumer

import fr.vetbrain.vetnutri_mp.Data.Labelable

/**
 * Énumération représentant les stades physiologiques d'un animal. Basée sur l'enum StadePhysio du
 * projet Java original.
 */
enum class StadePhysio(val categorie: Int, override val label: String) : Labelable {
    ADULTE(0, "Adulte"),
    CROISSANCE(1, "Croissance"),
    LACTATION(2, "Lactation"),
    GESTATION(3, "Gestation"),
    HOSPIT(4, "Hospit");

    companion object {
        /**
         * Récupère un stade physiologique à partir de son identifiant
         *
         * @param id L'identifiant à rechercher
         * @return Le stade physiologique correspondant ou ADULTE par défaut
         */
        fun getFromInt(id: Int): StadePhysio = entries.find { it.categorie == id } ?: ADULTE

        /**
         * Récupère un stade physiologique à partir de son nom
         *
         * @param name Le nom à rechercher
         * @return Le stade physiologique correspondant ou ADULTE par défaut
         */
        fun getFromString(name: String): StadePhysio = entries.find { it.label == name } ?: ADULTE

        /**
         * Récupère le nom d'un stade physiologique à partir de son identifiant
         *
         * @param id L'identifiant à rechercher
         * @return Le nom du stade physiologique correspondant ou "Adulte" par défaut
         */
        fun getStringFromInt(id: Int): String = getFromInt(id).label
    }

    /**
     * Convertit l'énumération en chaîne de caractères
     *
     * @return Le nom du stade physiologique
     */
    fun nameToString(): String = label

    override fun toString(): String = label
}
